src/main/webapp/js/searchableField.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Modified from the Jenkins combobox code
+ * https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/lib/form/combobox/combobox.js
+ *
+ * The Jenkins combobox only fills items 'onChange' of dependent fields. This means that if a combobox is dependent on
+ * itself (e.g a search field) then it is filled too late. This will fill the combobox options on 'input' and then
+ * when one is selected the human readable value will be in the combobox input and a computer-readable value will go
+ * in the associated value field.
+ **/
+Behaviour.specify(".searchable", 'searchableField', 200, function(el) {
+    var results = {};
+
+    new ComboBox(el, function(query) {
+        return Object.keys(results);
+    }, {});
+
+    el.addEventListener('input', function(e) {
+        var parameters = (el.getAttribute("fillDependsOn") || "").split(" ")
+            .reduce(function(params, fieldName) {
+                var dependentField = findNearBy(el, fieldName);
+                if (dependentField) {
+                    params[fieldName] = dependentField.value;
+                }
+                return params;
+            }, {});
+        new Ajax.Request(el.getAttribute("fillUrl"), {
+            parameters: parameters,
+            onSuccess: function(rsp) {
```

Comment:
We should add an `onFailure` as well and potentially indicate in the UI that we were unable to fetch responses.

Is this the canonical way async requests are done in Jenkins? jQuery _should_ be available via `var jQD = require('jquery-detached');` and then `var $ = jQD.getJQuery();` if you wanted to use `$.ajax({ ... })`. (That said, there's nothing wrong with using `Ajax.Request` if it gets the job done)